test/parallel_api/algorithm/alg.nonmodifying/transform_if.pass.cpp

```diff
@@ -0,0 +1,200 @@
+                        return false;
+                    }
+                    else if (i % 2 == 1 && *result != -(3 * j))
+                    { // reverse iterator case
+                        return false;
+                    }
+                }
+                else if (*mask == 0 && *result != 0)
+                {
+                    return false;
+                }
+            }
+            else
+            { // odd # of elements in output sequence
+                if (*mask == 1)
+                {
+                    if (i % 2 == 0 && (*result != -(3 * i) && *result != -(3 * j)))
+                    {
+                        return false;
+                    }
+                }
+                else if (*mask == 0 && *result != 0)
+                {
+                    return false;
+                }
+            }
+        }
+        return true;
+    }
+
+    template <typename Policy, typename InputIterator1, typename InputIterator2, typename OutputIterator, typename Size>
+    void
+    operator()(Policy&& exec, InputIterator1 first, InputIterator1 last, InputIterator2 mask,
+               InputIterator2 /* mask_end */, OutputIterator result, OutputIterator result_end, Size n)
+    {
+        using value_type1 = typename ::std::iterator_traits<InputIterator1>::value_type;
+        using value_type2 = typename ::std::iterator_traits<InputIterator2>::value_type;
+        // call transform_if
+        oneapi::dpl::transform_if(exec, first, last, mask, result, mutable_negate_first<value_type1, value_type2>{},
+                                  mutable_check_mask_second<value_type1, value_type2>{});
+
+        EXPECT_TRUE(check(first, last, mask, result, n), "transform_if binary wrong result");
+        // reset output elements to 0
+        ::std::fill(result, result_end, 0);
+    }
+};
+
+struct test_transform_if_unary
+{
+    template <typename It1, typename Out, typename Size>
+    bool
+    check(It1 first, It1 last, Out result, Size n)
+    {
+
+        for (; first != last; ++first, ++result)
+        {
+            if (*first % 3 == 0 && *result != -(*first))
+            {
+                return false;
+            }
+            else if (*first % 3 != 0 && *result != 0)
+            {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    template <typename Policy, typename InputIterator1, typename OutputIterator, typename Size>
+    void
+    operator()(Policy&& exec, InputIterator1 first, InputIterator1 last, OutputIterator result,
+               OutputIterator result_end, Size n)
+    {
+        using value_type1 = typename ::std::iterator_traits<InputIterator1>::value_type;
+        // call transform_if
+        oneapi::dpl::transform_if(exec, first, last, result, mutable_negate<value_type1>{},
+                                  mutable_check_mod3_is_0<value_type1>{});
+
+        EXPECT_TRUE(check(first, last, result, n), "transform_if unary wrong result");
```

Comment:
This approach ```EXPECT_TRUE(check(``` is correct.
But usually we use a little bit another approach:
- create expected data;
- compare them with actual data;
- show first difference and exit with special exit code.
For example, you may find the usage of ```EXPECT_EQ_N```, ```EXPECT_EQ_RANGES``` macros in tests.
In this test implementation we never see what is the diff and on which item.

Reply:
Sure, I will change the tests to this pattern.  Thanks.

Reply:
Done.